src/test/kotlin/io/snyk/plugin/ui/toolwindow/SnykAuthPanelTest.kt

```diff
@@ -0,0 +1,139 @@
+        val jButton = UIComponentFinder.getJButtonByText(cut, searchString)
+        assertNotNull(jButton)
+        assertEquals("Connect your IDE to Snyk", jButton!!.text)
+    }
+
+    @Test
+    fun `should display right authenticate button text when in test group`() {
+        every { project.service<AmplitudeExperimentService>().isPartOfExperimentalWelcomeWorkflow() } returns true
+
+        val cut = SnykAuthPanel(project)
+        val searchString = cut.authenticateButtonText()
+        val jButton = UIComponentFinder.getJButtonByText(cut, searchString)
+        assertNotNull(jButton)
+        assertEquals("Test code now", jButton!!.text)
+    }
+
+    @Test
+    fun `should display right description label when in control group`() {
+        every { project.service<AmplitudeExperimentService>().isPartOfExperimentalWelcomeWorkflow() } returns false
+
+        val cut = SnykAuthPanel(project)
+
+        val jLabel = UIComponentFinder.getJLabelByText(cut, "Please authenticate to Snyk and connect your IDE")
+        assertNotNull(jLabel)
+    }
+
+    @Test
+    fun `should display right description label when in test group`() {
+        every { project.service<AmplitudeExperimentService>().isPartOfExperimentalWelcomeWorkflow() } returns true
+        val expectedText = """
+        |<html><ol>
+        |  <li align="left">Authenticate to Snyk.io</li>
+        |  <li align="left">Analyze code for issues and vulnerabilities</li>
+        |  <li align="left">Improve you code and upgrade dependencies</li>
+        |</ol>
+        |</html>
+        """.trimMargin()
+
+        val cut = SnykAuthPanel(project)
+
+        val jLabel = UIComponentFinder.getJLabelByText(cut, expectedText)
+        assertNotNull(jLabel)
+    }
+
+    @Test
+    fun `should send tracking to amplitude on authenticate button press if in test group`() {
+        every { project.service<AmplitudeExperimentService>().isPartOfExperimentalWelcomeWorkflow() } returns true
+        stubAuthenticationButtonServiceInteractions()
+
+        val cut = SnykAuthPanel(project)
+        val authenticateButton = UIComponentFinder.getJButtonByText(cut, "Test code now")
+        assertNotNull(authenticateButton)
+
+        authenticateButton!!.action.actionPerformed(mockk())
+
+        verify { analyticsService.logAuthenticateButtonIsClicked(any()) }
+    }
+
+    @Test
+    fun `should send tracking to amplitude on authenticate button press if in control group`() {
+        every { project.service<AmplitudeExperimentService>().isPartOfExperimentalWelcomeWorkflow() } returns false
+        stubAuthenticationButtonServiceInteractions()
+
+        val cut = SnykAuthPanel(project)
+        val authenticateButton = UIComponentFinder.getJButtonByText(cut, "Connect your IDE to Snyk")
+        assertNotNull(authenticateButton)
+
+        authenticateButton!!.action.actionPerformed(mockk())
+
+        verify { analyticsService.logAuthenticateButtonIsClicked(any()) }
+    }
```

Comment:
tests above seems to be too experiment agnostic. will require extra work to remove experiment in the future...

Reply:
It is experiment agnostic, because we send that event independent of experiment for control AND test group.

Reply:
Tracking in this case != experiment